src/Entity/EntityId.php

```diff
@@ -30,6 +37,68 @@ public function getSerialization() {
 	}
 
 	/**
+	 * Returns an array with 3 elements: the foreign repository name as the first element, the local ID as the last
+	 * element and everything that is in between as the second element.
+	 *
+	 * @param $serialization
+	 * @return array
+	 */
+	public static function splitSerialization( $serialization ) {
+		$parts = explode( ':', $serialization );
+		$localPart = array_pop( $parts );
+		$repoName = array_shift( $parts ) ?: '';
+		$prefixRemainder = implode( ':', $parts );
+
+		return array( $repoName, $prefixRemainder, $localPart );
+	}
+
+	/**
+	 * Concatenates parts of an EntityId serialization with ':'.
+	 *
+	 * @param array $parts
+	 * @return string
+	 */
+	public static function joinSerialization( $parts ) {
+		return implode( ':', array_filter( $parts ) );
+	}
+
+	/**
+	 * Returns the foreign repository name or empty string for local entities.
+	 * @return string
+	 */
+	public function getRepoName() {
+		$parts = self::splitSerialization( $this->serialization );
+
+		return $parts[0];
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getLocalPart() {
+		$parts = self::splitSerialization( $this->serialization );
+
+		return self::joinSerialization( array( '', $parts[1], $parts[2] ) );
+	}
+
+	/**
+	 * @return bool
+	 */
+	public function isForeign() {
+		return strpos( $this->serialization, ':' ) > 0;
+	}
+
+	protected static function normalizeIdSerialization( $id ) {
+		$parts = self::splitSerialization( ltrim( $id, ':' ) );
+
+		return self::joinSerialization( array(
```

Comment:
The trimming of the leading ":" should not be tied to the upper-casing of the local part. The leading ":" must always be stripped, that's part of the convention for foreign repo prefixes, which use the same syntax for all IDs.
Making the local part upper case is tied to specific EntityId types, and should only be called by those that actually need that.